src/Core/src/Platform/Linux/MauiGtkApplication.cs

```diff
@@ -0,0 +1,225 @@
+using System;
+using System.Diagnostics;
+using Gdk;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Maui.Hosting;
+using Microsoft.Maui.LifecycleEvents;
+using Gtk;
+using Microsoft.Maui.Graphics;
+
+namespace Microsoft.Maui
+{
+
+	public class MauiGtkApplication<TStartup> : MauiGtkApplication
+		where TStartup : IStartup, new()
+	{
+
+		public void Run()
+		{
+			Launch(new EventArgs());
+		}
+
+		protected void RegisterLifecycleEvents(Gtk.Application app)
+		{
+
+			app.Startup += OnStartup!;
+			app.Shutdown += OnShutdown!;
+			app.Opened += OnOpened;
+			app.WindowAdded += OnWindowAdded;
+			app.Activated += OnActivated!;
+			app.WindowRemoved += OnWindowRemoved;
+			app.CommandLine += OnCommandLine;
+
+		}
+
+		protected void OnStartup(object sender, EventArgs args)
+		{
+
+			StartupMainWindow();
+
+			Services.InvokeLifecycleEvents<LinuxLifecycle.OnStartup>(del => del(CurrentGtkApplication, args));
+		}
+
+		protected void OnOpened(object o, GLib.OpenedArgs args)
+		{
+			Services?.InvokeLifecycleEvents<LinuxLifecycle.OnOpened>(del => del(CurrentGtkApplication, args));
+		}
+
+		protected void OnActivated(object sender, EventArgs args)
+		{
+			StartupLauch(sender, args);
+
+			Services?.InvokeLifecycleEvents<LinuxLifecycle.OnApplicationActivated>(del => del(CurrentGtkApplication, args));
+		}
+
+		protected void OnShutdown(object sender, EventArgs args)
+		{
+			Services?.InvokeLifecycleEvents<LinuxLifecycle.OnShutdown>(del => del(CurrentGtkApplication, args));
+
+			MauiGtkApplication.DispatchPendingEvents();
+
+		}
+
+		protected void OnCommandLine(object o, GLib.CommandLineArgs args)
+		{
+			;
+		}
+
+		protected void OnWindowRemoved(object o, WindowRemovedArgs args)
+		{
+			;
+		}
+
+		protected void OnWindowAdded(object o, WindowAddedArgs args)
+		{
+			;
```

Comment:
We will need this events to have notifications at cross platform Window level, but for now, we can remove it or just include a comment.